DigitalLearningSolutions.Data/DataServices/CourseDataService.cs

```diff
@@ -392,9 +392,17 @@ OUTPUT INSERTED.Id
 
             int supervisorDelegateId = (int)connection.ExecuteScalar(
                     @"SELECT COALESCE
-                 ((SELECT ID FROM SupervisorDelegates WHERE [SupervisorAdminId] = @supervisorId), 0) AS ID",
-                    new { supervisorId }
+                 ((SELECT TOP 1 ID FROM SupervisorDelegates WHERE SupervisorAdminID = @supervisorId AND CandidateID = @candidateId), 0) AS ID",
+                    new { supervisorId, candidateId }
                 );
+            if (supervisorDelegateId == 0)

```

Comment:
Added this to account for cases where the supervisor delegate record did not already exist.